src/Database/Redis/Cluster.hs

```diff
@@ -61,7 +62,7 @@ type IsReadOnly = Bool
 data Connection = Connection (HM.HashMap NodeID NodeConnection) (MVar Pipeline) (MVar ShardMap) CMD.InfoMap IsReadOnly
```

Comment:
Since a pool is now per node, the map "HM.HashMap NodeID NodeConnection" is probably never going to update.
Can we find a way to update this mapping also? This will fail requests when we scale up Redis cluster where we won't be able to find the new NodeID in this map.

Reply:
@ishan-juspay is it okay if we trigger the refresh logic at the point where some NodeId is not found in the map? We can have a Pool of a pair of HashMap and ShardMap and whenever some NodeId is not found in the HashMap, it will throw and Exception and a new constructor will be called for the Pair.

Reply:
And this way, if some Resources are working with NodeId that they are able to find, then they will keep working without any problems.

Reply:
We can have this HashMap as an IORef and update it whenever RefreshShardMap is called ?